analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -501,25 +486,39 @@ private const val PACKAGE_SEPARATOR = "# "
  */
 private const val WHY_CHUNK_SIZE = 32
 
-private fun getRevision(version: String): String {
-    version.withoutSuffix("+incompatible")?.let { return getRevision(it) }
-
-    PSEUDO_VERSION_REGEXES.forEach { regex ->
-        regex.find(version)?.let { matchResult ->
-            return matchResult.groups["sha1"]!!.value
-        }
-    }
-
-    return version
+/**
+ * The format of `.info` the Go command line tools cache under '$GOPATH/pkg/mod'.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+private data class ModuleInfoFile(
+    @JsonProperty("Origin")
+    val origin: Origin
+) {
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class Origin(
+        @JsonProperty("VCS")
+        val vcs: String? = null,
+        @JsonProperty("URL")
+        val url: String? = null,
+        @JsonProperty("Ref")
+        val ref: String? = null,
+        @JsonProperty("Hash")
+        val hash: String? = null,
+        @JsonProperty("Subdir")
+        val subdir: String? = null
+    )
 }
 
-internal fun GoMod.ModuleInfo.toVcsInfo(): VcsInfo {
-    val hostname = GITHUB_NAME_REGEX.matchEntire(path)?.let {
-        it.groupValues[1]
-    } ?: path
+private fun GoMod.ModuleInfo.toVcsInfo(): VcsInfo {
+    val info = jsonMapper.readValue<ModuleInfoFile>(File(goMod).resolveSibling("$version.info"))
+    if (info.origin.vcs != "git") return VcsInfo.EMPTY
 
-    val vcsInfo = VcsHost.parseUrl("https://$hostname")
-    return vcsInfo.copy(revision = getRevision(version))
+    return VcsInfo(
+        type = VcsType.GIT,
```

Comment:
Instead of hard-coding Git here and inline 514, consider something like

```
    ...
    val type = info.origin.vcs?.let { VcsType.forName(it) }.takeIf { it == VcsType.GIT } ?: return VcsInfo.EMPTY

    return VcsInfo(
        type = type,
        ...
```